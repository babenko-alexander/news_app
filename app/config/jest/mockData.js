export const newsArr = [
  {
  author: "Тори Путимцева",
  description: "Петра Порошенка в будь-якому випадку змусять прийти на допит в ГБР",
  id: "2019-07-17T12:54:00Zhttps://news.google.com/__i/rss/rd/articles/CBMiZmh0dHBzOi8vem5hai51YS9wb2xpdGljcy8yNDczODctZGJyLXBvb2JpY3lhbG8tc2lsb3l1LXByaXZvbG9rdGktcG9yb3NoZW5rYS15YWtzaGNoby1wcm9kb3Z6aGl0LWJpZ2F0adIBamh0dHBzOi8vYW1wLnpuYWoudWEvcG9saXRpY3MvMjQ3Mzg3LWRici1wb29iaWN5YWxvLXNpbG95dS1wcml2b2xva3RpLXBvcm9zaGVua2EteWFrc2hjaG8tcHJvZG92emhpdC1iaWdhdGk?oc=5ДБР пообіцяло силою приволокти Порошенка, якщо продовжить бігати - Znaj.ua",
  publishedAt: "2019-07-17   12:54:00",
  title: "ДБР пообіцяло силою приволокти Порошенка, якщо продовжить бігати - Znaj.ua",
  url: "https://news.google.com/__i/rss/rd/articles/CBMiZmh0dHBzOi8vem5hai51YS9wb2xpdGljcy8yNDczODctZGJyLXBvb2JpY3lhbG8tc2lsb3l1LXByaXZvbG9rdGktcG9yb3NoZW5rYS15YWtzaGNoby1wcm9kb3Z6aGl0LWJpZ2F0adIBamh0dHBzOi8vYW1wLnpuYWoudWEvcG9saXRpY3MvMjQ3Mzg3LWRici1wb29iaWN5YWxvLXNpbG95dS1wcml2b2xva3RpLXBvcm9zaGVua2EteWFrc2hjaG8tcHJvZG92emhpdC1iaWdhdGk?oc=5",
  urlToImage: "https://znaj.ua/images/2019/07/17/nMb3mO2ewMrQ3BD9Fd1NAkV2QdQwJk4v0jGNPZDW.jpeg"
  },
  {
  author: null,
  description: "Нові свідки у справі про авіакатастрофу рейсу МН17 зв'язалися з Міжнародною слідчою групою (JIT). — Укрінформ.",
  id: "2019-07-17T13:04:00Zhttps://news.google.com/__i/rss/rd/articles/CBMibWh0dHBzOi8vd3d3LnVrcmluZm9ybS51YS9ydWJyaWMtd29ybGQvMjc0MjI3MS11LXNwcmF2aS1taDE3LXphdmlsaXNhLW5vdmktc3ZpZGtpLW1pem5hcm9kbmEtc2xpZGNhLWdydXBhLmh0bWzSAXFodHRwczovL3d3dy51a3JpbmZvcm0udWEvYW1wL3J1YnJpYy13b3JsZC8yNzQyMjcxLXUtc3ByYXZpLW1oMTctemF2aWxpc2Etbm92aS1zdmlka2ktbWl6bmFyb2RuYS1zbGlkY2EtZ3J1cGEuaHRtbA?oc=5У справі MH17 з'явилися нові свідки — Міжнародна слідча група - Укрінформ. Новини України та світу",
  publishedAt: "2019-07-17   13:04:00",
  title: "У справі MH17 з'явилися нові свідки — Міжнародна слідча група - Укрінформ. Новини України та світу",
  url: "https://news.google.com/__i/rss/rd/articles/CBMibWh0dHBzOi8vd3d3LnVrcmluZm9ybS51YS9ydWJyaWMtd29ybGQvMjc0MjI3MS11LXNwcmF2aS1taDE3LXphdmlsaXNhLW5vdmktc3ZpZGtpLW1pem5hcm9kbmEtc2xpZGNhLWdydXBhLmh0bWzSAXFodHRwczovL3d3dy51a3JpbmZvcm0udWEvYW1wL3J1YnJpYy13b3JsZC8yNzQyMjcxLXUtc3ByYXZpLW1oMTctemF2aWxpc2Etbm92aS1zdmlka2ktbWl6bmFyb2RuYS1zbGlkY2EtZ3J1cGEuaHRtbA?oc=5",
  urlToImage: "https://static.ukrinform.com/photos/2016_09/thumb_files/630_360_1475079509-3067.jpg"
  },
  {
  author: null,
  description: "Президент повідомив, що кандидатуру на посаду голови Миколаївської ОДА усе ще шукають.",
  id: "2019-07-17T10:54:00Zhttps://news.google.com/__i/rss/rd/articles/CBMijQFodHRwczovL3d3dy51bmlhbi51YS9wb2xpdGljcy8xMDYyMDY1NC1jaGlub3ZuaWtpLXBvdm9keWF0c3lhLXlhay1iYW5kaXRpLXplbGVuc2tpeS1uYWthemF2LXYtby1nbGF2aS1taWtvbGFqaXZza29qaS1vZGEtenZpbG5pdGlzLXZpZGVvLmh0bWzSAQA?oc=5«Чиновники поводяться, як бандити»: Зеленський наказав в.о. глави Миколаївської ОДА звільнитись (відео) - УНІАН",
  publishedAt: "2019-07-17   10:54:00",
  title: "«Чиновники поводяться, як бандити»: Зеленський наказав в.о. глави Миколаївської ОДА звільнитись (відео) - УНІАН",
  url: "https://news.google.com/__i/rss/rd/articles/CBMijQFodHRwczovL3d3dy51bmlhbi51YS9wb2xpdGljcy8xMDYyMDY1NC1jaGlub3ZuaWtpLXBvdm9keWF0c3lhLXlhay1iYW5kaXRpLXplbGVuc2tpeS1uYWthemF2LXYtby1nbGF2aS1taWtvbGFqaXZza29qaS1vZGEtenZpbG5pdGlzLXZpZGVvLmh0bWzSAQA?oc=5",
  urlToImage: "https://images.unian.net/photos/2019_07/thumb_files/1000_545_1563360613-7722.jpg?1"
  }
];

export const newsArrT = [
  {
  author: "Ториasfasf",
  description: "Петра Порошеsaffasfasfнка в будь-якому випадку змусять прийти на допит в ГБР",
  id: "2019-07-17T12:54:00Zhttps://news.google.com/__i/rss/rd/articles/CBMiZmh0dHBzOi8vem5hai51YS9wb2xpdGljcy8yNDczODctZGJyLXBvb2JpY3lhbG8tc2lsb3l1LXByaXZvbG9rdGktcG9yb3NoZW5rYS15YWtzaGNoby1wcm9kb3Z6aGl0LWJpZ2F0adIBamh0dHBzOi8vYW1wLnpuYWoudWEvcG9saXRpY3MvMjQ3Mzg3LWRici1wb29iaWN5YWxvLXNpbG95dS1wcml2b2xva3RpLXBvcm9zaGVua2EteWFrc2hjaG8tcHJvZG92emhpdC1iaWdhdGk?oc=5ДБР пообіцяло силою приволокти Порошенка, якщо продовжить бігати - Znaj.ua",
  publishedAt: "2019-07-11   12:54:00",
  title: "ДБР пообasfsafіцяло силою приволокти Порошенка, якщо продовжить бігати - Znaj.ua",
  url: "https://news.google.com/__i/rss/rd/articles/CBMiZmh0dHBzOi8vem5hai51YS9wb2xpdGljcy8yNDczODctZGJyLXBvb2JpY3lhbG8tc2lsb3l1LXByaXZvbG9rdGktcG9yb3NoZW5rYS15YWtzaGNoby1wcm9kb3Z6aGl0LWJpZ2F0adIBamh0dHBzOi8vYW1wLnpuYWoudWEvcG9saXRpY3MvMjQ3Mzg3LWRici1wb29iaWN5YWxvLXNpbG95dS1wcml2b2xva3RpLXBvcm9zaGVua2EteWFrc2hjaG8tcHJvZG92emhpdC1iaWdhdGk?oc=5",
  urlToImage: "https://znaj.ua/images/2019/07/17/nMb3mO2ewMrQ3BD9Fd1NAkV2QdQwJk4v0jGNPZDW.jpeg"
  }
];

export const article =   {
  author: "Тори Путимцева",
  description: "Петра Порошенка в будь-якому випадку змусять прийти на допит в ГБР",
  id: "2019-07-17T12:54:00Zhttps://news.google.com/__i/rss/rd/articles/CBMiZmh0dHBzOi8vem5hai51YS9wb2xpdGljcy8yNDczODctZGJyLXBvb2JpY3lhbG8tc2lsb3l1LXByaXZvbG9rdGktcG9yb3NoZW5rYS15YWtzaGNoby1wcm9kb3Z6aGl0LWJpZ2F0adIBamh0dHBzOi8vYW1wLnpuYWoudWEvcG9saXRpY3MvMjQ3Mzg3LWRici1wb29iaWN5YWxvLXNpbG95dS1wcml2b2xva3RpLXBvcm9zaGVua2EteWFrc2hjaG8tcHJvZG92emhpdC1iaWdhdGk?oc=5ДБР пообіцяло силою приволокти Порошенка, якщо продовжить бігати - Znaj.ua",
  publishedAt: "2019-07-17   12:54:00",
  title: "ДБР пообіцяло силою приволокти Порошенка, якщо продовжить бігати - Znaj.ua",
  url: "https://news.google.com/__i/rss/rd/articles/CBMiZmh0dHBzOi8vem5hai51YS9wb2xpdGljcy8yNDczODctZGJyLXBvb2JpY3lhbG8tc2lsb3l1LXByaXZvbG9rdGktcG9yb3NoZW5rYS15YWtzaGNoby1wcm9kb3Z6aGl0LWJpZ2F0adIBamh0dHBzOi8vYW1wLnpuYWoudWEvcG9saXRpY3MvMjQ3Mzg3LWRici1wb29iaWN5YWxvLXNpbG95dS1wcml2b2xva3RpLXBvcm9zaGVua2EteWFrc2hjaG8tcHJvZG92emhpdC1iaWdhdGk?oc=5",
  urlToImage: "https://znaj.ua/images/2019/07/17/nMb3mO2ewMrQ3BD9Fd1NAkV2QdQwJk4v0jGNPZDW.jpeg"
  };
  