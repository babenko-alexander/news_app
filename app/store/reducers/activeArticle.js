export function setArticle(data) {
  return {
    type: 'SET_ARTICLE',
    data
  };
}

export default function activeArticle(state = null, action) {
  switch (action.type) {
    case 'SET_ARTICLE': {
      return action.data;
    }
    case 'RESET_ARTICLE':
      return state;
    default:
      return state;
  }
}