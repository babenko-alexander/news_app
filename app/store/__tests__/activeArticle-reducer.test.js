import activeArticle from '../reducers/activeArticle';
import { article } from '../../config/jest/mockData';


describe('activeArticle reducer', () => {
  it('returns the same state on an unhandled action', () => {
    expect(activeArticle([], {type: '_NULL'})).toMatchSnapshot();
  });
  it('handles SET_ARTICLE action', () => {
    expect(activeArticle([], {type: 'SET_ARTICLE', article})).toMatchSnapshot();
  });
  it('handles RESET_ARTICLE action', () => {
    expect(activeArticle(article, {type: 'RESET_ARTICLE'})).toMatchSnapshot();
  });
});
