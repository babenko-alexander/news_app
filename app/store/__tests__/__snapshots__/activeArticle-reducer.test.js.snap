// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`activeArticle reducer handles RESET_ARTICLE action 1`] = `
Object {
  "author": "Тори Путимцева",
  "description": "Петра Порошенка в будь-якому випадку змусять прийти на допит в ГБР",
  "id": "2019-07-17T12:54:00Zhttps://news.google.com/__i/rss/rd/articles/CBMiZmh0dHBzOi8vem5hai51YS9wb2xpdGljcy8yNDczODctZGJyLXBvb2JpY3lhbG8tc2lsb3l1LXByaXZvbG9rdGktcG9yb3NoZW5rYS15YWtzaGNoby1wcm9kb3Z6aGl0LWJpZ2F0adIBamh0dHBzOi8vYW1wLnpuYWoudWEvcG9saXRpY3MvMjQ3Mzg3LWRici1wb29iaWN5YWxvLXNpbG95dS1wcml2b2xva3RpLXBvcm9zaGVua2EteWFrc2hjaG8tcHJvZG92emhpdC1iaWdhdGk?oc=5ДБР пообіцяло силою приволокти Порошенка, якщо продовжить бігати - Znaj.ua",
  "publishedAt": "2019-07-17   12:54:00",
  "title": "ДБР пообіцяло силою приволокти Порошенка, якщо продовжить бігати - Znaj.ua",
  "url": "https://news.google.com/__i/rss/rd/articles/CBMiZmh0dHBzOi8vem5hai51YS9wb2xpdGljcy8yNDczODctZGJyLXBvb2JpY3lhbG8tc2lsb3l1LXByaXZvbG9rdGktcG9yb3NoZW5rYS15YWtzaGNoby1wcm9kb3Z6aGl0LWJpZ2F0adIBamh0dHBzOi8vYW1wLnpuYWoudWEvcG9saXRpY3MvMjQ3Mzg3LWRici1wb29iaWN5YWxvLXNpbG95dS1wcml2b2xva3RpLXBvcm9zaGVua2EteWFrc2hjaG8tcHJvZG92emhpdC1iaWdhdGk?oc=5",
  "urlToImage": "https://znaj.ua/images/2019/07/17/nMb3mO2ewMrQ3BD9Fd1NAkV2QdQwJk4v0jGNPZDW.jpeg",
}
`;

exports[`activeArticle reducer handles SET_ARTICLE action 1`] = `undefined`;

exports[`activeArticle reducer returns the same state on an unhandled action 1`] = `Array []`;
